torch/_export/non_strict_utils.py

```diff
@@ -91,6 +98,15 @@ def make_fake_inputs(nn_module, args, constraints):
     ) as fake_mode:
         original_signature = inspect.signature(nn_module.forward)
         params = original_signature.parameters
+
+        # This means we can't reliably rely on signature to get the source info
+        if len(params) != len(args):
+            params = []  # type: ignore[assignment]
+            count = 0
+            for arg in args:
+                params.append("arg_" + str(count))  # type: ignore[attr-defined]
+                count += 1
+
```

Comment:
I remember we discuessed offline that we should at least make the code simple here to have a single execution path. `len(params) != len(args)` is too arbitrary to use and you also need to consider the cases where default argument values and keyword arguments are present. 

Overall doesn't this show another pitfall of the current dynamic shape API design which is code-oriented instead data-oriented? (meaning we must rely on the information from source code rather than specify constraints on data directly)